### ISSUE TITLE
Loading round into chamber before fire the firearm

### ISSUE THREAD

**Issue by SaltGin** [CleverRaven/Cataclysm-DDA#34248]
**Is your feature request related to a problem? Please describe.**  
The firearm part of CDDA is pretty awesome to play with, but there is still a problem that kind of disappointed me for a long time. There should be a round in the chamber before you fire the firearm, instead of just load the magazine into the magwell and shoot.

Like Glock 19, there should be 15+0 rounds first after you insert the magazine, then you rack the slide, there should be 14+1 rounds in it, one in the chamber. After that, you can eject the mag and load a new round into the mag (or load a round into chamber first and insert a full magazine), so there could be actually 15+1 capacity for a Glock 19.

Most firearms can be fired by just loading one round into the chamber, but one thing you need to confirm is the firearm which have mag safety* won't fire without a magazine in the magwell. This feature is useful especially when you don't have proper magazine for a rifle or you need emergency-one-shot to deal with some fast monsters like zombie dog, you can lock open the bolt, insert the round into the chamber, release the bolt and shoot. Or you might just want an extra bullet in your gun.

*The mag safety is kind of common in the pistols but not in the rifles. 

**Describe the solution you'd like**  
Add an item to indicate the ammo that contains in the chamber and a general action that considered as charging the gun (like rack the pistol slide, pull the pump-action shotgun forend, the charging handle, or the bolt of the bolt-action firearms). This action is also useful for more realistic malfunction clearance, like, to eject misfired round, or clear the stovepipe (casing failed to eject)*.

When selected to reload a gun, the system should ask whether you want to load a round into the chamber or load the magazine into magwell.

*It would be more complicated to simulate different kinds of malfunctions, I will open a new issue to discuss.

**Comment by tenmillimaster** [comment]
Waiting on #32331 before implementation.

**Comment by anothersimulacrum** [comment]
See also #26730

**Comment by VReaperV** [comment]
> Add an item to indicate the ammo that contains in the chamber and a general action that considered as charging the gun (like rack the pistol slide, pull the pump-action shotgun forend, the charging handle, or the bolt of the bolt-action firearms). This action is also useful for more realistic malfunction clearance, like, to eject misfired round, or clear the stovepipe (casing failed to eject)*.

That would add more keypresses for guns that need to be rechambered manually. Plus you'd have to manually load the first round for other guns.

**Comment by thethunderhawk** [comment]
There’s almost no situation in CDDA where you’d want to load a magazine without chambering a round as well. You could make it automatic, with a unique action to unload a chambered round.

**Comment by ThomasLinkin** [comment]
It called Tactical Reload my dude. I opened a request like an hour ago and NotFuji decline it. He meant it wasn't worth the effort,so i suppose you should stop having a dream of this one like mine.

**Comment by kevingranade** [comment]
This has been discussed to death, if someone wants to implement it they can, but we're not leaving an issue open for it.
